src/contracts/Invitation/Persistence/Gateway.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Contracts\User\Invitation\Persistence;
+
+interface Gateway
+{
+    public function addInvitation(
+        string $email,
+        string $siteAccessName,
+        string $hash,
+        ?int $roleId = null,
+        ?int $userGroupId = null,
+        ?string $limitation = null,
+        ?string $limitationValue = null
+    ): array;
+
+    public function getInvitation(string $hash);
+
+    public function invitationExistsForEmail(string $email): bool;
+
+    public function getInvitationForEmail(string $email);
```

Comment:
```suggestion
    public function getInvitationByEmail(string $email);
```
like in InvitationService